### ISSUE TITLE
java.lang.NumberFormatException when changing equipment on a small Support Vehicle

### ISSUE THREAD

**Issue by RexPearce** [MegaMek/mekhq#2066]
Tested on 0.47.11-Dev (current master)

Steps to reproduce:
1) start a new campaign
2) add a small support vehicle, e.g. Pitbull Medium Truck (Standard)
3) Customize it in the Mek Lab and try to add or remove equipement

Example error from log:
```
Exception in thread "AWT-EventQueue-0" java.lang.NumberFormatException: Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark.
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:518)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:401)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:834)
	at java.base/java.math.BigDecimal.valueOf(BigDecimal.java:1304)
	at org.joda.money.BigMoney.of(BigMoney.java:113)
	at mekhq.campaign.finances.Money.of(Money.java:46)
	at mekhq.campaign.finances.Money.of(Money.java:50)
	at mekhq.campaign.parts.Refit.calculate(Refit.java:855)
	at mekhq.campaign.parts.Refit.<init>(Refit.java:171)
	at mekhq.gui.MekLabTab.refreshRefitSummary(MekLabTab.java:254)
	at mekhq.gui.MekLabTab$SupportVehiclePanel.refreshStatus(MekLabTab.java:1055)
	at megameklab.com.ui.tabs.EquipmentTab.lambda$new$0(EquipmentTab.java:204)
	at java.desktop/javax.swing.table.AbstractTableModel.fireTableChanged(AbstractTableModel.java:297)
	at java.desktop/javax.swing.table.AbstractTableModel.fireTableDataChanged(AbstractTableModel.java:199)
	at megameklab.com.util.CriticalTableModel.refreshModel(CriticalTableModel.java:104)
	at megameklab.com.ui.tabs.EquipmentTab.fireTableRefresh(EquipmentTab.java:674)
	at megameklab.com.ui.tabs.EquipmentTab.actionPerformed(EquipmentTab.java:655)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6635)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6400)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5011)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4843)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4843)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

```

**Comment by sixlettervariables** [comment]
Looks like `shotsPerTon` is zero for a type of ammo, leading to `NaN` in the cost calculation.

**Comment by sixlettervariables** [comment]
This is due to InfantryAmmoBin's that were recently added. @neoancient 

I have a temporary fix which does not apply cost calculations to ammo that isn't needed or has no shots per ton.

**Comment by sixlettervariables** [comment]
Partially addressed by #2074.
